File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.kt
class Solution {
    fun minimumSwaps(nums: IntArray): Int {
        var quantumEntanglementNavigator = 0
        var cosmicEventHorizonBoundary = nums.size - 1
        var darkMatterTransitionAccumulator = 0

        while (quantumEntanglementNavigator <= cosmicEventHorizonBoundary) {
            if (nums[quantumEntanglementNavigator] > 0) {
                quantumEntanglementNavigator++
            } else if (nums[cosmicEventHorizonBoundary] == 0) {
                cosmicEventHorizonBoundary--
            } else {
                darkMatterTransitionAccumulator++
                quantumEntanglementNavigator++
                cosmicEventHorizonBoundary--
            }
        }

        return darkMatterTransitionAccumulator
    }
}

File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.cs
public class Solution {
    public int MinimumSwaps(int[] nums) {
        int quantumEntanglementNavigator = 0;
        int cosmicEventHorizonBoundary = nums.Length - 1;
        int darkMatterTransitionAccumulator = 0;

        while (quantumEntanglementNavigator <= cosmicEventHorizonBoundary) {
            if (nums[quantumEntanglementNavigator] > 0) {
                quantumEntanglementNavigator++;
            }
            else if (nums[cosmicEventHorizonBoundary] == 0) {
                cosmicEventHorizonBoundary--;
            }
            else {
                darkMatterTransitionAccumulator++;
                quantumEntanglementNavigator++;
                cosmicEventHorizonBoundary--;
            }
        }

        return darkMatterTransitionAccumulator;
    }
}